core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/SailDatasetImpl.java

```diff
@@ -285,6 +286,39 @@ public void handleClose() throws SailException {
 		}
 	}
 
+	@Override
+	public CloseableIteration<? extends Triple, SailException> getTriples(Resource subj, IRI pred, Value obj)
+			throws SailException {
+		CloseableIteration<? extends Triple, SailException> iter;
+		if (changes.isStatementCleared()) {
+			iter = null;
+		} else {
+			iter = derivedFrom.getTriples(subj, pred, obj);
+		}
+
+		if (iter == null) {
+			return new EmptyIteration<>();
+		}
+		return iter; // TODO we will need to figure out a way to handle transaction isolation with deprecated and
+						// approved data
```

Comment:
FWIW I need to take a closer look at this to make sure, but it's possible that we don't need to handle this here, because it will be filtered out higher up the chain (when statements that use these triples as either subject or object are retrieved).